src/gsi/read_nsstbufr.f90

```diff
@@ -260,6 +279,31 @@ subroutine read_nsstbufr(nread,ndata,nodata,gstime,infile,obstype,lunout, &
            else
               sst = bmiss
            endif
+        elseif ( trim(subset) == 'NC031005' ) then            ! Argo or Glider
+           msst = 12.0_r_kind                                 ! for ARGO, assign to be 12
+           call ufbint(lunin,tpf3,3,65535,klev,'SSTH SALNH WPRES') ! get klev,T,water pres & salinity
+
+           call ufbint(lunin,r_rpid,1,1,iret,'WMOP')
+           write(crpid,'(I7)') int(r_rpid)
+
+           apres(:) = real(tpf3(3,:))
+
+           pres1 = minval(apres)/10000.  ! converting wpres from Pa to dbar for minmum depth 
+                                                  ! Not always layer 1 
+!
+!          get zob in meters
+!
+           zob = ((-3.434e-12*pres1+1.113e-7)*pres1+0.712953)*pres1 + 14190.7*log(1.0+1.83e-5*pres1)
+           zob = (zob/(980.+1.113e-4*pres1))*1000.
```

Comment:
Please add precision to real values -3.434e-12, 1.113e-7, 0.712953, 14190.7, 1.0, 1.83e-5, 980., 1.113e-4, and 1000.